lib/ansible/modules/cloud/azure/azure_rm_webapp_facts.py

```diff
@@ -0,0 +1,348 @@
+                    "use32_bit_worker_process": true,
+                    "virtual_applications": [
+                        {
+                            "physical_path": "site\\wwwroot",
+                            "preload_enabled": false,
+                            "virtual_path": "/"
+                        }
+                    ],
+                    "vnet_name": "",
+                    "web_sockets_enabled": false
+                },
+                "tags": {
+                    "tag1": "test"
+                },
+                "type": "Microsoft.Web/sites"
+            }]
+'''
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.common import AzureMissingResourceHttpError, AzureHttpError
+except:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+AZURE_OBJECT_CLASS = 'WebApp'
+
+info_level_spec = dict(
+    level=dict(
+        type='str',
+        choices=[
+            'basic',
+            'configuration',
+            'app_settings'
+        ],
+        default='basic'
+    )
+)
+
+
+class AzureRMWebAppFacts(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list'),
+            info_level=dict(
+                type='list',
+                elements='dict',
+                options=info_level_spec
+            )
+        )
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_webapps=[])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+        self.info_level = None
+
+        super(AzureRMWebAppFacts, self).__init__(self.module_arg_spec,
+                                                   supports_tags=False,
+                                                   facts_module=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.format == "curated":
+            self.fail('Not implemented.')
+
+        if not self.name and self.info_level:
+            self.fail('info_level must be specified with name parameter')
+
+        if self.name:
+            self.results['ansible_facts']['azure_webapps'] = self.list_by_name()
+
+            if self.info_level:
+                for level in self.info_level:
+                    if level["level"] == "configuration":
+                        self.results['ansible_facts']['azure_webapps'][0]['site_config'] = self.list_webapp_configuration()
+                    if level["level"] == "app_settings":
+                        self.results['ansible_facts']['azure_webapps'][0]['app_settings'] = self.list_webapp_appsettings()
+
+        elif self.resource_group:
+            self.results['ansible_facts']['azure_webapps'] = self.list_by_resource_group()
```

Comment:
recently the idea is not to place facts inside 'ansible_facts'

Reply:
then where to put it?  has the idea been agreed?

Reply:
yes, even mentioned today on the meeting... 
also I would rename 'azure_webapps' to just 'webapps', as azure part is a kind of obvious